src/components/slide/slide.tsx

```diff
@@ -1,27 +1,43 @@
 import React, {
   createContext,
-  useContext,
   useCallback,
-  useState,
+  useContext,
   useEffect,
-  useMemo
+  useMemo,
+  useState
 } from 'react';
 import ReactDOM from 'react-dom';
-import PropTypes from 'prop-types';
 import styled, { css, ThemeContext } from 'styled-components';
-import { background, color, space } from 'styled-system';
-import { DeckContext } from '../deck/deck';
-import { useSpring, animated } from 'react-spring';
+import {
+  background,
+  BackgroundProps,
+  color,
+  ColorProps,
+  space,
+  SpaceProps
+} from 'styled-system';
+import { DeckContext, SlideId, TemplateFn } from '../deck/deck';
+import { animated, useSpring } from 'react-spring';
 import { useSlide } from '../../hooks/use-slides';
-import { useCollectSteps } from '../../hooks/use-steps';
+import { ActivationThresholds, useCollectSteps } from '../../hooks/use-steps';
 import { GOTO_FINAL_STEP } from '../../hooks/use-deck-state';
 import { useSwipeable } from 'react-swipeable';
+import { SlideTransition } from '../transitions';
 
 const noop = () => {};
 
-export const SlideContext = createContext(null);
+export const SlideContext = createContext<{
+  immediate: boolean;
+  slideId: SlideId;
+  isSlideActive: boolean;
+  activationThresholds: ActivationThresholds;
+  activeStepIndex: number;
+}>(null);
```

Comment:
Does it make sense to make this an `export type`

Reply:
Yes, makes total sense.  Extracted this to `SlideContextType`, and did the same for `DeckContextType`.